airflow/executors/kubernetes_executor.py

```diff
@@ -607,6 +607,7 @@ def try_adopt_task_instances(self, tis: List[TaskInstance]) -> List[TaskInstance
             for pod in pod_list.items:
                 self.adopt_launched_task(kube_client, pod, pod_ids)
         self._adopt_completed_pods(kube_client)
+        self._handle_zombied_istio_pods(kube_client)
```

Comment:
This does _work_, but it does take an extra adoption cycle before it actually handles these pods. Probably fine.